source/simulator/assemblers/newton_stokes.cc

```diff
@@ -148,6 +148,11 @@ namespace aspect
                                    :
                                    1;
 
+              // pre-compute the tensor contractions
+              std::vector<SymmetricTensor<2,dim> > tensor_contraction(stokes_dofs_per_cell);
```

Comment:
that's maybe not a good name for the variable. How about `deta_deps_times_eps_times_phi_j`? That's a long name, but at least it's descriptive. Other suggestions also welcome :-)

Reply:
I agree we need a better name, but we do not just use it for j, but also for i. So something like `viscosity_derivative_wrt_strain_rate_x_grads_phi_u_x_strain_rate` or `eta_derivative_wrt_eps_x_grads_phi_u_x_eps` or `eta_derivative_wrt_eps_times_grads_phi_u_times_eps` or just `deta_deps_times_eps_times_phi`